extension/vector/src/index/hnsw_index_utils.cpp

```diff
@@ -61,41 +61,84 @@ void HNSWIndexUtils::validateColumnType(const catalog::TableCatalogEntry& tableE
     validateColumnType(type);
 }
 
-double HNSWIndexUtils::computeDistance(MetricType funcType, const float* left, const float* right,
-    uint32_t dimension) {
+template<auto Func, typename T>
+static double computeDistance(const void* left, const void* right, uint32_t dimension) {
     double distance = 0.0;
-    switch (funcType) {
+    Func(static_cast<const T*>(left), static_cast<const T*>(right), dimension, &distance);
```

Comment:
Is there a specific reason why we use templated function pointer there? (I mean `template<auto Func`)
I suggest passing the FUNC as a function pointer instead of a template value.
1. Using template function pointers can increase the binary size. The compiler generates a definition for each possible FUNC.
2. Using function pointers is more flexible than function template.

Reply:
The reason for using template here is because there is no common function signature for these simsimd distance functions unless we wrap them up again, which I want to avoid by templating here.